### PR TITLE
Namespace

### PR DESCRIPTION
**Pull request by mrnetops**
* Set default namespace & subsystem (fastly and rt respectively)
* Some README.md enhancements

I think we should remove the namespace and subsystem options entirely, but I figure I'll start here ;)

### CODE REVIEW

**Comment by peterbourgon** [comment]
Tthe defaults provided in the binary itself (blank) should be the defaults in the Docker image, too.

File: README.md [hunk 1]
```diff
@@ -51,9 +55,10 @@ VERSION
   2.0.0
 ```
 
-A valid Fastly API -token is mandatory. [See this link][token] for information
-on creating API tokens. Providing individual -service IDs is optional. Service
-IDs are available at the top of your [Fastly dashboard][db]. 
+A valid Fastly API `-token` is mandatory. [See this link][token] for information on creating API tokens. 
+
+Optional `-service` IDs can be specified to limit monitoring to specific
+services.  Service IDs are available at the top of your [Fastly dashboard][db].
```
Comment (peterbourgon) [inline README.md]:
```suggestion
services. Service IDs are available at the top of your [Fastly dashboard][db]. 
```

File: README.md [hunk 2]
```diff
@@ -51,9 +55,10 @@ VERSION
   2.0.0
 ```
 
-A valid Fastly API -token is mandatory. [See this link][token] for information
-on creating API tokens. Providing individual -service IDs is optional. Service
-IDs are available at the top of your [Fastly dashboard][db]. 
+A valid Fastly API `-token` is mandatory. [See this link][token] for information on creating API tokens.
```
Comment (peterbourgon) [inline README.md]: Wrap at 80 cols please.

File: main.go [hunk 3]
```diff
@@ -26,8 +26,8 @@ func main() {
 		token      = fs.String("token", "", "Fastly API token (required)")
 		serviceIDs = stringslice{}
 		addr       = fs.String("endpoint", "http://127.0.0.1:8080/metrics", "Prometheus /metrics endpoint")
-		namespace  = fs.String("namespace", "", "Prometheus namespace (optional)")
-		subsystem  = fs.String("subsystem", "", "Prometheus subsystem (optional)")
+		namespace  = fs.String("namespace", "fastly", "Prometheus namespace (optional)")
+		subsystem  = fs.String("subsystem", "rt", "Prometheus subsystem (optional)")
```
Comment (peterbourgon) [inline main.go]: -1, please leave them blank.

Comment (mrnetops) [reply main.go]: So, this one is the meat of it.  I'm a little unclear what functionality we're looking to offer by not providing a default namespace.

even the core prometheus bindings provide fixed namespaces

go_
process_
promhttp_

I would also reference https://prometheus.io/docs/instrumenting/writing_exporters/#naming
```
Metric names for applications should generally be prefixed by the exporter name, e.g. haproxy_up.
```
Could you possibly clarify what use case you are looking to cover?

I'm pushing on this one as ideally I'd like to build up to making canned grafana dashboards available to folks, but that becomes problematic if there is not a standard for the metric names (which includes a standard and default namespace and subsystem)

**Comment by peterbourgon** [comment]
Thanks for the link. I’m convinced. `fastly_rt_` works, or maybe `fastly_realtime_`?

**Review by peterbourgon — APPROVED** [review]

**Pull request merged**